lib/routes/prototype-admin-routes.js

```diff
@@ -283,16 +287,36 @@ router.post('/templates/install', async (req, res) => {
     res.redirect(url)
   }
 
+  if (!chosenUrl.length) {
+    renderError('missing')
+    return
+  }
+
+  if (chosenUrl === '/') {
+    renderError('singleSlash')
+    return
+  }
+
   if (chosenUrl[0] !== '/') {
     renderError('slash')
     return
   }
 
+  if (chosenUrl[chosenUrl.length - 1] === '/') {
+    renderError('endsWithSlash')
+    return
+  }
+
   if (await fse.exists(installLocation)) {
     renderError('exists')
     return
   }
 
+  if (!chosenUrl.match(/^(\/[a-zA-Z0-9-_]+)+(\.)?$/)) {
```

Comment:
I think this regex is saying the path may end with a full stop, what's the thinking behind that?

Also, this stop people from using letters with accents or from alphabets other than English, is that restriction necessary or can we be more permissive?

Reply:
I'll fix, I forgot to remove that bit in the regex

Reply:
The other bit can be worked out when we pair later today

Reply:
agree that we're in danger of restricting valid urls - I think we should err on permissive - ideally whatever's allowed in a url